aip/general/0162.md

```diff
@@ -342,67 +252,39 @@ ranges of time.
 
-```proto
-rpc DeleteBookRevision(DeleteBookRevisionRequest)
-    returns (Book) {
-  option (google.api.http) = {
-    delete: "/v1/{name=publishers/*/books/*}:deleteRevision"
-  };
-}
-```
+- Resource revisions **should** be automatically deleted on parent resource deletion.
+- If resource revisions meant to have longer lifespan than the parent resource, resource and resource revision
+should be side by side resource collections.
 
-```proto
-message DeleteBookRevisionRequest {
-  // The name of the book revision to be deleted, with a revision ID explicitly
-  // included.
-  //
-  // Example: publishers/123/books/les-miserables@c7cfa2a8
-  string name = 1 [
-    (google.api.field_behavior) = REQUIRED,
-    (google.api.resource_reference) = {
-      type: "library.googleapis.com/Book"
-    }];
-}
-```
+## Rationale
 
-- The request message **must** have a `name` field to identify the resource
-  revision being deleted.
-  - The explicit revision ID **must** be required (the method **must** fail
-    with `INVALID_ARGUMENT` if it is not provided, and **must not** default to
-    the latest revision).
-  - The field **should** be [annotated as required][aip-203].
-  - The field **should** identify the [resource type][aip-123] that it
-    references.
-- The API **must not** overload the `DeleteBook` method to serve both purposes
-  (this could lead to dangerous or confusing mistakes).
-- If the resource supports soft delete, then revisions of that resource
-  **should** also support soft delete.
-- The method **must not** cause the _resource_ to be deleted.
-  - Because a resource **should not** exist with zero revisions, the method
-    **must** fail with `FAILED_PRECONDITION` if the user attempts to delete the
-    only revision.
-- The response message **must** be the resource itself. There is no
-  `DeleteBookRevisionResponse`.
-  - The response **should** include the fully-populated resource unless it is
-    infeasible to do so.
-
-### Character Collision
-
-Most resource names have a restrictive set of characters, but some are very
-open. For example, Google Cloud Storage allows the `@` character in filenames,
-which are part of the resource name, and therefore uses the `#` character to
-indicate a revision.
-
-APIs **should** avoid permitting the `@` character in resource names, and if
-APIs that do permit it need to support resources with revisions, they
-**should** pick an appropriate separator depending on how and where the API is
-used, and **must** clearly document it.
+### Deprecate the @-based approach
+
+Historically, Resource revision was recommended to be referenced by `{resource_id}@{revision_id}`.
+
+It has several disadvantages:
+
+-  List revisions is a custom method (:listRevisions) on the resource collection
+-  Delete revision is a custom method on the resource collection
+-  Not visible in api discovery doc
+-  Resource ID cannot use `@`
```

Comment:
It might be worth listing the benefits that we're losing as well though. In particular, the ability to handle a resource revision or a normal resource in the same way, both in terms of code and resource names.

I'm pretty sure we'll want to add the ability to specify that a resource name is "either X or Y" where the main use case is "either a resource or a resource revision".

Reply:
It's possibly also worth noting that although this makes revisions feel like "normal" resources in some ways, there are still significant differences between them and regular ones - e.g. creation patterns.

Reply:
FWIW I think the creation could also be normalized now - a tag on a revision is a skip and a hop away from a resource ID.

I think it's fair to include the advantages, as well as explain how they might achieved in the new world. For now I think we're talking about a "*" resource reference since we don't have the semantics to express references to more than one resource.